File: src/main/java/com/borvalcode/sudoku/infrastructure/delivery/armeria/controller/InsertController.java
package com.borvalcode.sudoku.infrastructure.delivery.armeria.controller;

import com.borvalcode.sudoku.core.entity.Insert;
import com.borvalcode.sudoku.core.usecase.InsertValue;
import com.linecorp.armeria.common.HttpResponse;
import com.linecorp.armeria.server.annotation.Post;

public class InsertController {

  private final InsertValue insertValue;

  public InsertController(InsertValue insertValue) {
    this.insertValue = insertValue;
  }

  @Post("/sudoku/insert")
  @JsonPost
  public HttpResponse insert(Insert insert) {
    return Controller.process(this.insertValue, insert);
  }
}

File: src/main/kotlin/com/borvalcode/sudoku/infrastructure/delivery/armeria/controller/InsertController.kt
package com.borvalcode.sudoku.infrastructure.delivery.armeria.controller

import com.borvalcode.sudoku.core.entity.Insert
import com.borvalcode.sudoku.core.usecase.InsertValue
import com.linecorp.armeria.common.HttpResponse
import com.linecorp.armeria.server.annotation.Post

class InsertController(private val insertValue: InsertValue) {

    @Post("/sudoku/insert")
    @JsonPost
    fun insert(insert: Insert): HttpResponse = Controller.process(insertValue, insert)
}